Automattic/wp-calypso
# New User Survey: Adds a new user survey for new user signups right after they signup

Related to pau2Xa-5AZ-p2

## Proposed Changes 

### Show a post signup survey

* Adds a new user survey as an experiment to fulfil a customer data gathering exercise
* The experiment is loaded on load of the `start` page. The experiment is loaded only if the flow is `onboarding` and the user is not currently signed up. 
https://github.com/Automattic/wp-calypso/blob/4c61fe774e6cfa3a197243f2e5488ba48079cec3/client/signup/controller.js#L237-L248

* The survey will be excluded in all other instances and the `control` variant.
* The data will be submitted as tracks props

<img width="1381" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3422709/d71d7690-c533-41bd-9c58-4980c6add8d3">

### Shuffle options logic

When the form is visible, we shuffle options if the variant is `treatment_scramble`. If not the order will remain static. When shuffling the order we record it in state and provide this information as a tracks prop for data analysis.


## Testing Instructions


Assign yourself to the variant in  21652-explat-experiment
* Try with a fresh user on the onboarding flow `/start` the survey view should be visible
* Try with a fresh user on the any other signup flow `/start` the survey view should NOT be visible
* Try with an existing user (user with at least 1 active site) start logged out and go to `/start` survey should not be visible
* With a fresh user fill the survey make sure they are accurately reflected when submitted.

Assign yourself to the control in  21652-explat-experiment
* Try with a fresh user on the onboarding flow `/start` the survey view should NOT be visible

Use feature flag 

* This can also be tested via the feature flags onboarding/new-user-survey, onboarding/new-user-survey-scrambled. 
Eg: ...
   - `/start?flags="onboarding/new-user-survey"` 
   - `/start?flags="onboarding/new-user-survey-scrambled"`

## Pre-merge Checklist
